packages/playwright-test/src/loader.ts

```diff
@@ -131,6 +131,7 @@ export class Loader {
     this._fullConfig.workers = takeFirst(this._configOverrides.workers, config.workers, baseFullConfig.workers);
     this._fullConfig.webServer = takeFirst(this._configOverrides.webServer, config.webServer, baseFullConfig.webServer);
     this._fullConfig._plugins = takeFirst(this._configOverrides.plugins, config.plugins, baseFullConfig._plugins);
+    this._fullConfig.metadata = mergeObjects(this._configOverrides.metadata, mergeObjects(config.metadata, baseFullConfig.metadata));
```

Comment:
`metadata` is an `any`-typed thing in our public API. We can't change that. So, better hope this doesn't try to merge a number and an object or array!

Not sure what to do here…other then maybe some nicer error message in the case of explosions. reviewers: wdyt?